dietpi/dietpi-software

```diff
@@ -9047,6 +9047,11 @@ location ^~ /rutorrent {
 	auth_basic "ruTorrent login";
 	auth_basic_user_file /etc/.rutorrent-htaccess;
 	location ~ ^/rutorrent/(?:conf|share)(?:/|$) { return 404; }
+
+	location ~ \.php$ {
+		include snippets/fastcgi-php.conf;
+		fastcgi_pass unix:/var/run/php/php-fpm.sock;
+	}
```

Comment:
The `php` handler is defined in `nginx.conf` and assured to point to the correct socket file. `/run/php/php-fpm.sock` does not exist on Buster yet, and generally `/var/run` is deprecated, `/run` should be used 😉.
```suggestion
	location ~ \.php(?:$|/) {
		include snippets/fastcgi-php.conf;
		fastcgi_pass php;
	}
```

Reply:
Done. May I ask what does the "(?:$|/)" do? I barely touch this section so I always put php$ instead.

Reply:
Did you answer your original question yourself? `(?:$|/)` is a regular expression which matches either `$` (for URLs which end with the PHP file name) or `/` (for URLs which have a path after the PHP file name). Not sure whether it is required in ruTorrent (yet), but since it's common to use paths elements after the `index.php` for internal handling, I just use this for every PHP handler block.